### ISSUE TITLE
Office 365 incoming webhook connector for Teams is deprecated, Workflow connector for incoming webhook does not work with notifications

### ISSUE THREAD

**Issue by JimReprogle** [opensearch-project/dashboards-notifications#220]
What is the bug?
Microsoft is deprecating the Office 365 connectors for teams, and will require users to utilize the workflow app to post adaptive cards to Teams channels via webhook.

How can one reproduce the bug?

Go to 'Notifications->Channels->Create Channel'
Give a name to your channel
Select Channel Type of Microsoft Teams
Enter the webhook URL from the Workflow app (not the O365 webhook connector URL. It should look something like:
[https://prod-XX.westus.logic.azure.com:443/workflows/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX](https://prod-xx.westus.logic.azure.com/workflows/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX)
Click "Send Test Message"
See the error: [status_exception] Wrong Microsoft Teams url. Should contain "webhook.office.com"
What is the expected behavior?
Expected behavior is that the call to the webhook will be able to send a message to the webhook URL. However, the URL is different and does not contain "webhook.office.com", and the expected type of data sent to the webhook is an adaptive card.

What is your host/environment?

OS: MacOS
Version: Sonoma 14.5

Do you have any additional context?
Microsoft is deprecating the type of webhook currently used in OpenSearch notifications sent to Microsoft Teams. They will require users to use the Workflow app to receive incoming webhooks, which in turn requires Openseaerch to send adaptive cards.

For reference: https://devblogs.microsoft.com/microsoft365dev/retirement-of-office-365-connectors-within-microsoft-teams/

**Comment by JimReprogle** [comment]
This can be worked around by utilizing the Custom Webhook notification in OpenSearch Dashboards.

**Comment by dcopestake** [comment]
> This can be worked around by utilizing the Custom Webhook notification in OpenSearch Dashboards.

Yep. I've commented with an example of how to do this on a similar issue: https://github.com/opensearch-project/notifications/issues/942#issuecomment-2291530022
